src/gui/shell.cpp

```diff
@@ -802,15 +615,31 @@ void Shell::wheelEvent(QWheelEvent *ev)
 	m_nvim->neovimObject()->vim_input(inp.toLatin1());
 }
 
+/// Resize remote Neovim (pixel coordinates)
+///
+/// The given size is rounded down to the nearest
+/// row/column count.
 void Shell::resizeNeovim(const QSize& newSize)
 {
-	uint64_t cols = newSize.width()/neovimCellWidth();
-	uint64_t rows = newSize.height()/neovimRowHeight();
+	uint64_t n_cols = newSize.width()/cellSize().width();
+	uint64_t n_rows = newSize.height()/cellSize().height();
+	resizeNeovim(n_cols, n_rows);
+}
 
-	// Neovim will ignore simultaneous calls to ui_try_resize
-	if (!m_resizing && m_nvim && m_attached &&
-			(cols != m_cols || rows != m_rows) ) {
-		m_nvim->neovimObject()->ui_try_resize(cols, rows);
+/// Resize remote Neovim.
+///
+/// Neovim ignores concurrent resizes. If you call this method while
+/// a resize is already in progress, the later call is delayed until
+/// the previous one is finished.
+void Shell::resizeNeovim(uint64_t n_cols, uint64_t n_rows)
+{
+	if (!m_nvim || (n_cols == columns() && n_rows == rows())) {
```

Comment:
/home/ico/public/git/neovim-qt/src/gui/shell.cpp:636:25: warning: comparison between signed and unsigned integer expressions [-Wsign-compare]
      if (!m_nvim || (n_cols == columns() && n_rows == rows())) {
                             ^
    /home/ico/public/git/neovim-qt/src/gui/shell.cpp:636:48: warning: comparison between signed and unsigned integer expressions [-Wsign-compare]
      if (!m_nvim || (n_cols == columns() && n_rows == rows())) {
                                                    ^

Conversion warnings on Arch with gcc 5.3.0.